stat/distmat/permutation_test.go

```diff
@@ -0,0 +1,62 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distmat
+
+import (
+	"testing"
+
+	"golang.org/x/exp/rand"
+
+	"gonum.org/v1/gonum/floats"
+	"gonum.org/v1/gonum/mat"
+)
+
+func TestUniformPermutation(t *testing.T) {
+	up := NewUniformPermutation(rand.NewSource(1))
+	for _, n := range []int{10, 32, 64, 100} {
+		m := mat.NewDense(n, n, nil)
+		if m == nil {
+			t.Error("Matrix failed")
+		}
+		up.PermTo(m)
+		if !confirmEachRowAndColumnOneNZ(m) {
+			t.Error("didnt get back a permutation matrix")
+		}
+	}
+
+}
+
+func confirmEachRowAndColumnOneNZ(m mat.Matrix) bool {
+	r, c := m.Dims()
```

Comment:
You may want to test that `r == c` here, to make this function independent from how the test calling it is written. Up to you.